docs/pages/blog/making-customizable-components.md

```diff
@@ -0,0 +1,287 @@
+:::
+
+This request makes sense.
+When you have a lot of columns, sorting them can make it easier to browse the list.
+But how should we implement this kind of customization?
+
+Adding a prop called `filterPanelColumnInputSortingStrategy` could work, but please don’t do that.
+It just doesn’t scale.
+There are too many different properties that developers might need to modify.
+You will end up with API documentation so long that it will take an eternity to scroll to the end—so nobody will read it.
+
+<img src="/static/blog/making-customizable-components/bruce.gif" style="width: 500px; margin-top: 16px; margin-bottom: 8px;" alt="Your user opening the list of props" />
+
+Here are a few better solutions that can scale more efficiently with complex components.
+
+### Other solutions
+
+#### Don’t bother with components
+
+Passing all the parameters as props of a single component does not work.
+So just don’t create components.
+
+It’s not a joke—that’s the approach of headless libraries such as react-table.
+Instead of providing working components, they only provide hooks for managing the features, and let developers build their components on top of that.
+
+If you’re willing to start from scratch, that's a nice approach.
+Use one hook to manage filtering, another one to manage the sorting, and then build your UI using returned values.
+
+This approach can scale because you can scope parameters to individual features.
+The filtering hooks will only take into account parameters impacting the filtering, and so on—so you can split your code feature by feature.
+
+But because this is a fully custom approach, it will take the most amount of work relative to all other options to construct a functional UI.
+If your main priority is to get up and running quickly, then this may not be a viable solution.
+
+#### Subdivide your components
+
+Another approach I like is to provide subcomponents.
+This is what we do for MUI Core components such as the [Menu](https://mui.com/material-ui/react-menu/).
+
+This is also the approach used by [react-admin](https://marmelab.com/react-admin/) to provide a customizable administration interface.
+Here is their quick start example.
+The idea is to put the `Admin` component at the root level of the app.
+It is a provider that’s responsible for managing all data fetching and passing that data back to components.
+
+The second important component is `ListGuesser` which defines how the data should be displayed.
+
+```jsx
+import React from 'react';
+import { Admin, Resource, ListGuesser } from 'react-admin';
+import simpleRestProvider from 'ra-data-simple-rest';
+
+const dataProvider = simpleRestProvider('https://domain.tld/api');
+
+const App = () => (
+  <Admin dataProvider={dataProvider}>
+    <Resource name="users" list={ListGuesser} />
+  </Admin>
+);
+export default App;
```

Comment:
I think that it would be better to use the convention we have to create components

```suggestion
export default function App() {
  return (
    <Admin dataProvider={dataProvider}>
      <Resource name="users" list={ListGuesser} />
    </Admin>
  );
}
```

we use this everywhere in the demos, it might make readers feel more "at home"